### PR TITLE
Add amazon EC2 Provider to IaaS

### PR DESCRIPTION
**Pull request by gtsalles**
This PR adds support for usage of EC2 from AWS to gofn, simply instantiating a new provider with secret and access keys.

Example to verify execution: 
https://github.com/gofn/gofn/tree/improvements/aws_ec2/examples/amazonec2

ref #54

### CODE REVIEW

File: iaas/amazonec2/amazonec2.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+			driverOpts.Values[f.String()] = false
+		}
+	}
+	// TODO: receive this configs to remove hard coded values
+	driverOpts.Values["swarm-host"] = "tcp://0.0.0.0:3376"
+	driverOpts.Values["swarm-image"] = "swarm:latest"
+	driverOpts.Values["swarm-strategy"] = "spread"
+	driverOpts.Values["swarm-discovery"] = ""
+	driverOpts.Values["swarm-master"] = false
+	driver.SetConfigFromFlags(&driverOpts)
```
Comment (golangci) [inline iaas/amazonec2/amazonec2.go]: Error return value of `driver.SetConfigFromFlags` is not checked

File: iaas/amazonec2/amazonec2.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	// https://github.com/docker/machine/blob/master/commands/create.go#L354-L361
+	for _, f := range flags {
+		driverOpts.Values[f.String()] = f.Default()
+
+		// Hardcoded logic for boolean... :(
+		if f.Default() == nil {
+			driverOpts.Values[f.String()] = false
+		}
+	}
+	// TODO: receive this configs to remove hard coded values
```
Comment (felipeweb) [inline iaas/amazonec2/amazonec2.go]: why have this todo ?

Comment (gtsalles) [reply iaas/amazonec2/amazonec2.go]: It's not the ideal scenario to have this configurations hard coded, instead, we should receive them as parameters, or (my preferred option) get the default values just like the docker-machine binary uses when creating a machine via cli, but honestly, I wasn't able to replicate this behavior.

Comment (felipeweb) [inline iaas/amazonec2/amazonec2.go]: you can implement functional option pattern so this values can be override easier. An exemple https://github.com/nuveo/storing/blob/master/awss3/awss3.go

Comment (gtsalles) [reply iaas/amazonec2/amazonec2.go]: I don't think I need to override this values, the better approach is just replicate the same default values that docker-machine uses. Right now, they are hard coded, the ideal would be retrieve them somehow, maybe something like the `driver.GetCreateFlags()` method, but retrieving the swarm values.

**Pull request merged**